### DISCUSSION TITLE
ArchetypeGenerator doesn't import projects

### DISCUSSION THREAD

**Discussion by kjsmita6** [eclipse-m2e/m2e-core#1096] [Q&A]
Hello, the javadoc for `ArchetypeGenerator.createArchetypeProjects()` says "Creates project structure using Archetype and then imports created project(s)". However, the code for importing seems to have been removed in https://github.com/eclipse-m2e/m2e-core/commit/c76ec876c69518c98e5bd3c4d09ee1713a41d020#diff-4742ec5a022dec0edc843e783c76a0a258224fcb273fb08ba3f107e0dc6887cbL139.

Is it importing the projects another way? Calling `createArchetypeProjects` and then `IWorkspaceRoot.getProject().exists()` returns `false`. `MavenProjectInfo` and `LocalProjectScanner` are written using `java.io.File` so I can't imagine it's workspace-aware.

Please let me know the answer or if we are still supposed to do `createArchetypeProjects` and then import them using `ProjectConfigurationManager.importProjects`. Thanks.

**Comment by kjsmita6** [comment]
For context, this is happening because we have a wizard that creates projects. The wizard creates a total of 3 projects: an aggregator project which contains two sub-projects. I am upgrading to the newest m2e release and this is happening because of the change to updating project configurations automatically:
![image (1)](https://user-images.githubusercontent.com/9405706/204360040-3012044a-2bfb-4d76-bded-23cd5e58a0c1.png)

**Comment by laeubi** [comment]
Archetype generation and importing projects are actually different topic. Because of this this was split into two steps see 

https://github.com/eclipse-m2e/m2e-core/blob/aa212c4901aedf93115f3ec642aa7ded62212046/org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/wizards/MavenProjectWizard.java#L244-L254

**Reply by kjsmita6** [reply]
Right, which is why I was saying that the javadoc needs to be updated 😉

**Reply by laeubi** [reply]
Just open a PR and suggest a change, but keep in mind that `ArchetypeGenerator` is **internal API** so other plugins should actually not depend on it! 

Making this accessible to others is part of this change, so you probably like to participate there:
- https://github.com/eclipse-m2e/m2e-core/pull/922
